src/library/transform/scan.cpp

```diff
@@ -1,147 +1,64 @@
-/* ************************************************************************
```

Comment:
I actually have no idea what is the legal status of this file because the code is adopted from bolt library.